akka-docs-dev/rst/java/code/docs/http/javadsl/server/directives/MethodDirectivesExamplesTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.http.javadsl.server.directives;
+
+import java.util.Arrays;
+
+import akka.http.javadsl.model.HttpMethod;
+import akka.http.javadsl.model.HttpMethods;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.headers.Host;
+import akka.http.javadsl.server.*;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+
+import org.junit.Test;
+
+public class MethodDirectivesExamplesTest extends JUnitRouteTest {
+  @Test
+  public void testDelete() {
+    //#delete
+    final Route route = delete(complete("This is a DELETE request."));
+
+    testRoute(route).run(HttpRequest.DELETE("/")).assertEntity(
+        "This is a DELETE request.");
+    //#delete
+  }
+
+  @Test
+  public void testGet() {
+    //#get
+    final Route route = get(complete("This is a GET request."));
+
+    testRoute(route).run(HttpRequest.GET("/")).assertEntity(
+        "This is a GET request.");
+    //#get
+  }
+
+  @Test
+  public void testHead() {
+    //#head
+    final Route route = head(complete("This is a HEAD request."));
+
+    testRoute(route).run(HttpRequest.HEAD("/")).assertEntity(
+        "This is a HEAD request.");
+    //#head
+  }
+
+  // FIXME where is HttpRequest.OPTIONS ?
+  //  @Test
+  //  public void testOptions() {
+  //    //#options
+  //    final Route route = options(complete("This is a OPTIONS request."));
+  //
+  //    testRoute(route).run(HttpRequest.OPTIONS("/")).assertEntity(
+  //        "This is a OPTIONS request.");
+  //    //#options
+  //  }
+
+  // FIXME where is HttpRequest.PATCH ?
+  //  @Test
+  //  public void testPatch() {
+  //    //#patch
+  //    final Route route = patch(complete("This is a PATCH request."));
+  //
+  //    testRoute(route).run(HttpRequest.PATCH("/", "patch content")).assertEntity(
+  //        "This is a PATCH request.");
+  //    //#patch
+  //  }
```

Comment:
The request method itself is `akka.http.javadsl.model.HttpMethods.PATCH` etc.  @patriknw